### PR TITLE
DevOps Flatpak revolution

### PR DESCRIPTION
**Pull request by nahuelwexd**
I had no idea what to title, so I put that on.

This PR corrects some errors that caused the action not to work correctly. Also, it converts the action from being a Docker Action to being a Javascript Action, that way it is able to start a Docker container with the necessary permissions so that Flatpak can compile at full speed with fuse activated.

I took the liberty of making some modifications to the entrypoint.sh as well, to add the new proposals from the DevOps with Flatpak Initiative that is currently used in the GNOME apps on GitLab. As a side effect of this, this action will no longer clone the repository from the URL provided in the manifest, but will instead use the repo from which it is invoked. The idea is that this is also combined with a caching of the `.flatpak-builder/{git,downloads}` folders to prevent the CI from slowing down with unnecessary download tasks.

I am currently using a small copy that I made for my own project, and it works perfectly. This copy is distributed in:
- Docker image: https://github.com/nahuelwexd/flatpak-docker
- GitHub Action: https://github.com/nahuelwexd/flatpak-builder-action
- Project it is working on: https://github.com/nahuelwexd/unitube-gtk

Note: Another side effect of this change is that now you will be required to create specific Docker image tags for each of the platforms: FreeDesktop, GNOME, KDE, etc. since this will no longer download the runtimes from Flathub

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+/***/ }),
+
+/***/ 888:
+/***/ (function(__unusedmodule, __unusedexports, __webpack_require__) {
+
+const core = __webpack_require__(369)
+const { spawn } = __webpack_require__(129)
+
+const docker_args = ['run', '--cap-add', 'SYS_ADMIN', '--cap-add',
```
Comment (bilelmoussaoui) [inline dist/index.js]: We don't need all of this, just set the seccomp profile to https://github.com/flatpak/flatpak/commit/26ad9154c31a484062ffcb46a3818a39ae35a866

**Comment by bilelmoussaoui** [comment]
Wow great job =) I will look into the rest of the code as soon as I can.

File: dist/index.js (hunk 1 above)
Comment (nahuelwexd) [reply dist/index.js]: I tried this locally but is not working. `--security-opt seccomp=flatpak-docker-seccomp.json` and the container is not able to mount the fuse fs

Comment (bilelmoussaoui) [reply dist/index.js]: we don't need fuse anymore, you can disable it by running flatpak-builder with ` --user --disable-rofiles-fuse`

Comment (nahuelwexd) [reply dist/index.js]: Right, but as stated by alexlarsson [here](https://github.com/flatpak/flatpak/issues/647#issuecomment-288063982) that would slow down the builds

Comment (bilelmoussaoui) [reply dist/index.js]: we can keep it as it's for now and figure out how to deal with that later.

**Comment by bilelmoussaoui** [comment]
Do you have any idea why the test build fails?

**Comment by nahuelwexd** [comment]
> As a side effect of this, this action will no longer clone the repository from the URL provided in the manifest, but will instead use the repo from which it is invoked.

The test build should now be in a full repository of an app, not here

**Comment by nahuelwexd** [comment]
Well, I added a little project that should make the tests pass, but you need to update the Docker image or they will still failing

**Comment by bilelmoussaoui** [comment]
Right, make sense. Thanks a lot! so we need to provide docker images for each runtime? :) that's a lot of images to maintain, I'm wondering if we should automate that using a github action too

**Comment by nahuelwexd** [comment]
> so we need to provide docker images for each runtime?
Yes, it also will avoid the need to download the runtimes every single time

**Comment by bilelmoussaoui** [comment]
Makes sense, what I thought of is keeping the dockerfile's here and just use a github action to build and publish them. To make it easier to contribute to the action/images :)

**Pull request merged**